labs/RivePlayer/src/StateMachineInput.cs

```diff
@@ -0,0 +1,117 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace CommunityToolkit.Labs.WinUI;
+
+// This base class wraps a custom, named state machine input value.
+public abstract partial /*???*/ class StateMachineInput : DependencyObject
+{
+    private string? _target;
+    public string? Target
+    {
+        get => _target;  // Must be null-checked before use.
+        set
+        {
+            _target = value;
+            Apply();
+        }
+    }
+
+    private WeakReference<RivePlayer> _rivePlayer = new WeakReference<RivePlayer>(null!);
+    protected WeakReference<RivePlayer> RivePlayer => _rivePlayer;
+
+    // Sets _rivePlayer to the given rivePlayer object and applies our input value to the state
+    // machine. Does nothing if _rivePlayer was already equal to rivePlayer.
+    internal void SetRivePlayer(WeakReference<RivePlayer> rivePlayer)
+    {
+        _rivePlayer = rivePlayer;
+        Apply();
+    }
+
+    protected void Apply()
+    {
+        if (!String.IsNullOrEmpty(_target) && _rivePlayer.TryGetTarget(out var rivePlayer))
+        {
+            Apply(rivePlayer, _target!);
+        }
+    }
+
+    // Applies our input value to the rivePlayer's state machine.
+    // rivePlayer and inputName are guaranteed to not be null or empty.
+    protected abstract void Apply(RivePlayer rivePlayer, string inputName);
+}
+
+[ContentProperty(Name = nameof(Value))]
+public class BoolInput : StateMachineInput
+{
+    // Define "Value" as a DependencyProperty so it can be data-bound.
+    public static readonly DependencyProperty ValueProperty = DependencyProperty.Register(
+        nameof(Value),
+        typeof(bool),
+        typeof(BoolInput),
+        new PropertyMetadata(false, OnValueChanged)
+    );
+
+    public bool Value
+    {
+        get => (bool)GetValue(ValueProperty);
+        set => SetValue(ValueProperty, value);
+    }
+
+    private static void OnValueChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+    {
```

Comment:
This is the method I expect to get called from the data binding.

If I manually call `rivePlayer.StateMachineInputCollection[0].Value = …`, this does fire. So I believe the StateMachineInputs are getting properly built into the RivePlayer's StateMachineInputCollection from the XAML side, and the listeners are firing from their `ValueProperty`.